### DISCUSSION TITLE
Weekly Sync #34 - Aug 25, 2021

### DISCUSSION THREAD

**Discussion by andresin87** [SUI-Components/sui-components#1651] [Weekly Sync Streaming]
[Notes from our last Weekly meeting](https://github.com/SUI-Components/sui-components/discussions/1641)

**Agenda**

- useSunscreen()
- [Dashboard](https://pages.github.mpi-internal.com/scmspain/design-systems/sui/index.html)
- [Contributor docs](https://github.com/SUI-Components/sui-components/tree/master/contributor-docs)
- Tokens ([see](https://github.com/turolopezsanabria/tokens)) [discussion](https://github.com/SUI-Components/sui-components/discussions/1570)
- [SUI-Tests metrics](https://github.com/SUI-Components/sui-components/tree/add/test-coverage#-test-coverage)
	see: https://github.com/SUI-Components/sui-components/commit/5196ed1c09bfdedd1e7db70d32677f5d2deb7754
- PinInput and Validation Code [discussion](https://github.com/SUI-Components/sui-components/discussions/1574)
- Avatar Badge
- switch single label positioning

**Comment by andresz1** [comment]
## Topics suggestions

### Documentation
Create a documentation website and use sui-studio as a development tool only. This will:
- Improve the visibility of our open source tools
- Improve documentation itself
- Avoid repeated docs (demo + readme)

Maybe we can use:
- JAMStack - Next.js or Gatsby and [MDX](https://mdxjs.com/)
- [docusaurus](https://docusaurus.io/)
- [docz](https://www.docz.site/)

Missing points that we can add:
- Colors
- Typography
- Spacing
- Theming
- Among others

Some examples:
- [baseui](https://baseweb.design/)
- [chakra-ui](https://chakra-ui.com/)
- [react-query](https://react-query.tanstack.com/)
- [react-hooks-form](https://react-hook-form.com/)

### Style system
Check what changes we can make to improve our style system (Sass). This will:
- Improve DX
- Improve build times (sass-loader is [slow](https://stackoverflow.com/questions/61475998/sass-loader-is-slow-in-webpack))
- Reduce used DOM Nodes

Somo ideas:
- Use css variables
- Move from Sass to another solution like Shopify did (see [thread](https://github.com/Shopify/polaris/discussions/44)). They are moving from Sass to [vanilla-extract](https://vanilla-extract.style/)
- Box component

```jsx
<Box my={2}>
  <Button mr={4}>
    Click me!
  </Button>

  <Typography as="p" variant="p1">
    hello
  </Typography>
</Box>
```

### Types
Maybe we can start adding [TypeScript](https://www.typescriptlang.org/) support like we are doing in [sui-tools](https://github.com/SUI-Components/sui). This will:
- Improve DX
- Reduce possible typing bugs

###  Components API
Use a compound components based approach to create components. This will:
- Increase readability and flexibility
- Reduce changes
- Improve DX

Most of our current components API is like:

```jsx
<Modal
  isOpen={isOpen}
  onClose={onClose}
  title="Title"
  closeIcon={<Icon />}
/>
```

When it could be something like:

```jsx
<Modal isOpen={isOpen} onClose={onClose}>
  <ModalHeader>Title</ModalHeader>
  
  <ModalCloseButton />

  <ModalContent>
    <Lorem count={2} />
  </ModalContent>

  <ModalFooter>
    <Button onClick={onClose}>
      Close
    </Button>
  </ModalFooter>
</Modal>
```

We could also set some agreements and naming conventions so getting used to the components API would be easier. For example, `boolean` props always prefixed with `is`. 

Somo examples:
- [react-material-ui](https://material-ui.com/)
- [theme-ui](https://theme-ui.com/)
- [chakra-ui](https://chakra-ui.com/)

### CSS Reset
Reduce browser inconsistencies in things like default line heights, margins and font sizes of headings, and so on. This will:
- Reduce resets repeated over and over again
- Single reset definition

This is how [chakra-ui]() does it because uses CSSinJS:
```jsx
<CSSReset />
```

We could have just a file instead and import it:
```css
@import '~@s-ui/sui-components/lib/reset';
```

### Accessibility
We should make our components usable by as many people as possible and try to provide first class UX, in order to do so we could:
- Analyse component by component alongside the design team
- Integrate tools like [axe](https://github.com/dequelabs/axe-core) and integrate them into our CI if possible
- Create task/issue defining the problems of each component (see [example](https://github.com/SUI-Components/sui-components/issues/1344))

**Reply by andresin87** [reply]
`@import '~@s-ui/sui-components/lib/reset';`

take a look to the

`https://github.com/SUI-Components/sui/blob/master/packages/sui-theme/src/_reset.scss`
